cmd/director_serve.go

```diff
@@ -66,6 +67,13 @@ func serveDirector( /*cmd*/ *cobra.Command /*args*/, []string) error {
 		return err
 	}
 
+	// The director needs its own private key. If one doesn't exist, this will generate it
+	issuerKeyFile := param.IssuerKey.GetString()
+	err = config.GeneratePrivateKey(issuerKeyFile, elliptic.P521())
```

Comment:
Due to limitations in other ares of Pelican, we can't support `elliptic.P521`. Can you bump this down to `elliptic.P256`?

A second note is that we also generate a private key in the func `generateTLSCertIfNeeded`. Can you remove it from there.

A third note is that ALL servers need a private key to work correctly. I vote we just remove this code from the Director (and from the origin and namespace registry, if it appears there) and instead move the key generation into `config.InitServer`. Then all servers will be guaranteed to generate keys in the same way.

Reply:
For this part, I will remove the code from generateTLSCertIfNeeded and change the encryption method. For the third note, I will open a separate issue for it.

Reply:
Looking at `namespace_registry_serve.go` which uses `generateTLSCertIfNeeded`, there's also a duplicate private key generation: https://github.com/PelicanPlatform/pelican/blob/e7fc4aa86870acd444e3f43b7388bba6a99748db/cmd/namespace_registry_serve.go#L47 and https://github.com/PelicanPlatform/pelican/blob/e7fc4aa86870acd444e3f43b7388bba6a99748db/cmd/namespace_registry_serve.go#L54 And the encryption there is also `P521`. 

With that, I will restructure the code in `director_serve.go` so that logic from `generateTLSCertIfNeeded` is pulled to `serveDirector` and I will move `generateTLSCertIfNeeded` to `namespace_registry_serve.go` just so that it won't break anything. I will open another issue for consolidating/fixing encryption in`namespace_registry_serve.go`

Reply:
Can you open another issue for switching any remaining `elliptic.P521` curves to `elliptic.P256`? I thought we had caught them all.